src/zerocash/Coin.cpp

```diff
@@ -73,7 +47,8 @@ Coin::Coin(const std::string bucket, Address& addr): addr_pk(), cm(), rho(ZC_RHO
     this->rho = rho_v;
     this->addr_pk = addr.getPublicAddress();
 
-    std::vector<unsigned char> a_pk = addr.getPublicAddress().getPublicAddressSecret();
+    std::vector<unsigned char> a_pk(addr.getPublicAddress().getPublicAddressSecret().begin(),
+                                    addr.getPublicAddress().getPublicAddressSecret().end());
```

Comment:
Actually this could be ``auto a_pk = from_uint256(addr.getPublicAddress().getPublicAddressSecret());``.